### PR TITLE
setup.cfg: Replace dashes with underscores

### PR DESCRIPTION
**Pull request by arthurzam**
## Description

Remove `description-file` from `setup.cfg`

## Motivation and Context

Setuptools v54.1.0 introduces a warning that the use of dash-separated options in 'setup.cfg' will not be supported in a future version. https://github.com/pypa/setuptools/commit/a2e9ae4cb

> UserWarning: Usage of dash-separated 'description-file' will not be supported in future versions. Please use the underscore name 'description_file' instead

## How Has This Been Tested?

Package builds successfully in tree and in Gentoo's building facilities.

## Change log

```
Remove unused description-file from setup.cfg
```

## Checklist:

- [X] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/gsd/blob/trunk-patch/CONTRIBUTING.rst).
- [X] I agree with the terms of the [**GSD Contributor Agreement**](https://github.com/glotzerlab/gsd/blob/trunk-patch/ContributorAgreement.md).
- [X] My name is on the list of contributors (`doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
Use the pull request template in the PR description, it includes steps that must be taken before pull requests may be accepted into this project: https://github.com/glotzerlab/gsd/blob/trunk-patch/.github/PULL_REQUEST_TEMPLATE.md

Regarding the proposed change, it would appear that `long-description` is actually an unused quantity propagated from tutorials by copy and paste: https://stackoverflow.com/questions/60084128/does-description-file-in-setup-cfg-section-metadata-have-any-effect. The actual setting that takes effect is in setup.py: https://github.com/glotzerlab/gsd/blob/12d131fc5b7f2890eb993a9c26dcd2c09b679a34/setup.py#L33 so we could instead remove `long-description` from `setup.cfg` entirely.

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [metadata]
-description-file = README.md
+description_file = README.md
```
Comment (joaander) [inline setup.cfg]: Remove this section instead.

**Review by joaander — CHANGES_REQUESTED** [review]
Remove the unused `description-file` instead of renaming it.
Also, apply the pull request template to your PR description and follow the instructions in it.

**Comment by arthurzam** [comment]
Thanks @joaander . I updated the PR description (hope it is correct now - sorry for missing the PR template).
Also took your changes and applied them (removed the `metadata` section).

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**